packages/compiler/src/experimental/typekit/kits/tuple.ts

```diff
@@ -0,0 +1,54 @@
+import { Tuple, Type } from "../../../core/types.js";
+import { defineKit } from "../define-kit.js";
+
+export interface TupleDescriptor {
+  /**
+   * The values for the tuple.
+   */
+  values?: Type[];
+}
+
+/**
+ * @experimental
+ */
+export interface TupleKit {
+  /**
+   * Check if a type is a tuple.
+   */
+  is(type: Type): type is Tuple;
+  /**
+   * Creates a tuple from a descriptor.
+   *
+   * @param desc The descriptor of the tuple. Passing no descriptor creates an empty tuple.
+   * @returns The tuple type.
+   */
+  create(desc?: TupleDescriptor): Tuple;
```

Comment:
I would also take an overload that just takes the types as array directly

Reply:
Added, although now I question if we need the TupleDescriptor overload at all - we can add it later if we actually need more properties or add it now to stay consistent with the rest of typekits. For now I left both here, but happy to remove one